src/base/static/state/ducks/map.js

```diff
@@ -1,311 +1,581 @@
-    id: layerId,
-    loadStatus: "unloaded",
-  },
-});
-export const setLayerLoaded = layerId => ({
-  type: SET_LAYER_LOAD_STATUS,
-  payload: {
-    id: layerId,
-    loadStatus: "loaded",
-  },
-});
-export const setLayerLoading = layerId => ({
-  type: SET_LAYER_LOAD_STATUS,
-  payload: {
-    id: layerId,
-    loadStatus: "loading",
-  },
-});
-export const setLayerError = layerId => ({
-  type: SET_LAYER_LOAD_STATUS,
-  payload: {
-    id: layerId,
-    loadStatus: "error",
-  },
-});
-export const setBasemap = basemapId => ({
-  type: SET_BASEMAP,
-  payload: basemapId,
-});
-export const setMapPosition = mapPosition => {
+}
+
+export function removeFocusedGeoJSONFeatures() {
   return {
-    type: SET_MAP_POSITION,
-    payload: mapPosition,
+    type: REMOVE_FOCUSED_GEOJSON_FEATURES,
+    payload: null,
   };
-};
-export const setMapSizeValidity = isValidSize => {
+}
+
+export function updateFocusedGeoJSONFeatures(features) {
   return {
-    type: SET_MAP_SIZE_VALIDITY,
-    payload: isValidSize,
+    type: UPDATE_FOCUSED_GEOJSON_FEATURES,
+    payload: features,
   };
-};
-export const activateFeatureFilter = filterInfo => {
+}
+
+export function updateLayerGroupVisibility(layerGroupId, isVisible) {
   return {
-    type: ACTIVATE_FEATURE_FILTER,
-    payload: filterInfo,
+    type: UPDATE_LAYER_GROUP_VISIBILITY,
+    payload: { layerGroupId, isVisible },
   };
-};
-export const deactivateFeatureFilter = filterInfo => {
+}
+
+export function updateSourceLoadStatus(sourceId, loadStatus) {
   return {
-    type: DEACTIVATE_FEATURE_FILTER,
-    payload: filterInfo,
+    type: UPDATE_SOURCE_LOAD_STATUS,
+    payload: { sourceId, loadStatus },
   };
-};
-export const resetFeatureFilterGroup = filterInfo => {
+}
+
+export function updateMapViewport(viewport) {
   return {
-    type: RESET_FEATURE_FILTER_GROUP,
-    payload: filterInfo,
+    type: UPDATE_VIEWPORT,
+    payload: { viewport, scrollZoomAroundCenter: false },
   };
-};
-export const updateMapDragged = isMapDragged => {
+}
+
+export function updateMapViewportFromReactMapGL(
+  viewport,
+  scrollZoomAroundCenter = false,
+) {
+  // We have a special action creator for viewport updates that originate from
+  // react-map-gl. The reason for this is that react-map-gl seems to cache
+  // the width and height of the map container at the beginning of a
+  // transition. If the viewport change that initiated the transition also
+  // changed the width and/or height (for example when a Place is clicked and
+  // the map resizes to make way for the content panel), then react-map-gl will
+  // immediately undo the width and height change. The result of this is yet
+  // another version of the off-center bug.
+  //
+  // So, we strip out the height and width from any viewport changes
+  // originating from react-map-gl. There should never be a situation where
+  // react-map-gl should be setting the width or height of the map container.
+  const { width, height, ...rest } = viewport;
```

Comment:
I think we need the other selector that keeps `width` and `height`, because that selector is used in case the viewport dimensions change (such as during a `resize` event, or when the content panel opens).

Reply:
I don't think I follow - which selector are you referring to? I left a new comment in MainMap that should clarify the changes I am thinking about...